fdbserver/DataDistributionQueue.actor.cpp

```diff
@@ -1484,16 +1658,184 @@ ACTOR static Future<bool> rebalanceTeams(DDQueueData* self,
 	    ShardsAffectedByTeamFailure::Team(sourceTeam->getServerIDs(), primary));
 	for (int i = 0; i < shards.size(); i++) {
 		if (moveShard == shards[i]) {
-			traceEvent->detail("ShardStillPresent", true);
-			self->output.send(RelocateShard(moveShard, priority));
+			self->output.send(RelocateShard(moveShard, priority, RelocateReason::REBALANCE_DISK));
 			return true;
 		}
 	}
 
-	traceEvent->detail("ShardStillPresent", false);
+	traceEvent->detail("SkipReason", "ShardNotPresent");
 	return false;
 }
 
+ACTOR Future<Void> getSrcDestTeams(DDQueueData* self,
+                                   int teamCollectionIndex,
+                                   GetTeamRequest srcReq,
+                                   GetTeamRequest destReq,
+                                   Reference<IDataDistributionTeam>* sourceTeam,
+                                   Reference<IDataDistributionTeam>* destTeam,
+                                   int priority,
+                                   TraceEvent* traceEvent) {
+
+	std::pair<Optional<Reference<IDataDistributionTeam>>, bool> randomTeam =
+	    wait(brokenPromiseToNever(self->teamCollections[teamCollectionIndex].getTeam.getReply(destReq)));
+	traceEvent->detail("DestTeam",
+	                   printable(randomTeam.first.map<std::string>(
+	                       [](const Reference<IDataDistributionTeam>& team) { return team->getDesc(); })));
+
+	if (randomTeam.first.present()) {
+		*destTeam = randomTeam.first.get();
+		std::pair<Optional<Reference<IDataDistributionTeam>>, bool> loadedTeam =
+		    wait(brokenPromiseToNever(self->teamCollections[teamCollectionIndex].getTeam.getReply(srcReq)));
+
+		traceEvent->detail("SourceTeam",
+		                   printable(loadedTeam.first.map<std::string>(
+		                       [](const Reference<IDataDistributionTeam>& team) { return team->getDesc(); })));
+
+		if (loadedTeam.first.present())
+			*sourceTeam = loadedTeam.first.get();
+	}
+	return Void();
+}
+
+ACTOR Future<Void> BgDDLoadRebalance(DDQueueData* self, int teamCollectionIndex, int ddPriority) {
+	state double rebalancePollingInterval = SERVER_KNOBS->BG_REBALANCE_POLLING_INTERVAL;
+	state int resetCount = SERVER_KNOBS->DD_REBALANCE_RESET_AMOUNT;
+	state Transaction tr(self->cx);
+	state double lastRead = 0;
+	state bool skipCurrentLoop = false;
+	state Future<Void> delayF = Never();
+	state const bool readRebalance = !isDiskRebalancePriority(ddPriority);
+	state const char* eventName =
+	    isMountainChopperPriority(ddPriority) ? "BgDDMountainChopper_New" : "BgDDValleyFiller_New";
```

Comment:
That's a good point. We don't have a way to enforce the priorities to be distinct. and the DD itself assumes the priorities should be distinct for each purpose. Otherwise, it also can't handle `priority_relocations` correctly. 
I prefer to leave it as it is now, then finds a way to enforce the distinct value in the near future's DD refractor work. The reason is now disk rebalance actors are the old ones, and even though MC and VF are the same in reading rebalance, it equals MC- or VF-only mode that I assume just changes the converge speed.